public/compute/the_hairball.md

```diff
@@ -72,23 +236,93 @@ In the current implementation, values are first taken from parameter files. If t
 if these values can be known at run-time, by analysing all previous steps of the protocol, where values are unknown.
 If values cannot be known at run-time, compute will give a generation error. 
 
+### Protocol headers
+#### Inputs and Outputs
+Each Protocol lists its Input parameters and its Output values in the header
+
+
+```
+#string project
+#string dir
+#list sample
+#output out1
+#output out2
+```
+
+This header would declare that this Protocol depends on the values of three Input parameters named `project`, `dir`, and `sample` and that when it is run it produces two Output values, namely `out1` and `out2`.
```

Comment:
This header states that this Protocol depends on the values of three Input parameters named `project`, `dir`, and `sample` and that when it is run it produces two Output values, namely `out1` and `out2`.